test/saml2.coffee

```diff
@@ -459,6 +463,46 @@ describe 'saml2', ->
         assert.deepEqual response, expected_response
         done()
 
+    it 'correctly parses an AuthnStatement with no session_index', (done) ->
```

Comment:
not sure i get what you mean here. If i understand correctly, you mean if you dont pass the require_session_index option it should fail? Right now the idea i was going with was to keep the default functionality as is and only make it optional if you pass `{require_session_index: false}` so default is `true`.